build.rs

```diff
@@ -1,33 +1,103 @@
+#[cfg(all(windows, target_env = "msvc"))]
+extern crate winreg;
+
 use std::process::Command;
-use std::path::Path;
+use std::path::{Path, PathBuf};
 use std::env;
 
-fn main() {
-    if cfg!(not(target_os = "windows")) {
-        return;
-    }
+#[cfg(not(windows))]
+fn main() { }
 
+#[cfg(windows)]
+fn main() {
     let out_dir = env::var("OUT_DIR").unwrap();
+    compile_resource(&out_dir);
+    println!("cargo:rustc-link-search=native={}", out_dir);
+}
+
+#[cfg(all(windows, target_env = "msvc"))]
+fn compile_resource(out_dir: &str) {
+    // `.res`es are linkable under MSVC as well as normal libraries.
+
+    fn find_windows_sdk_bin_dir() -> Option<PathBuf> {
+        use winreg::enums::*;
+
+        #[derive(Clone, Copy)]
+        enum Arch {
+            X86,
+            X64,
+        }
+
+        // Windows 8 - 10
+        fn find_windows_kits_bin_dir(key: &str, arch: Arch) -> Option<PathBuf> {
+            let res = winreg::RegKey::predef(HKEY_LOCAL_MACHINE)
+                .open_subkey_with_flags("SOFTWARE\\Microsoft\\Windows Kits\\Installed Roots", KEY_QUERY_VALUE)
```

Comment:
You can use a raw string literal here like `r"SOFTWARE\Microsoft\Windows Kits\Installed Roots"`